evm/src/main/java/org/hyperledger/besu/evm/operation/VirtualOperation.java

```diff
@@ -34,7 +34,10 @@ public VirtualOperation(final Operation delegate) {
   @Override
   public OperationResult execute(final MessageFrame frame, final EVM evm) {
     OperationResult result = delegate.execute(frame, evm);
-    return new OperationResult(result.getGasCost(), result.getHaltReason(), 0);
+    if (result.getHaltReason() != null) {
+      return result;
+    }
+    return new OperationResult(result.getGasCost(), 0);
```

Comment:
why we need to create a new one in this case and not returning result ?

Reply:
I think it was like that before because we don't want to increment the ProgramCounter so it's fixed to 0